classes/task/files.php

```diff
@@ -63,30 +63,26 @@ public static function spawn_shards(\tool_advancedreplace\files $record): void {
         $numshards = $record->get('shards');
         $timestart = time();
 
-        // Get the total number of files.
-        $criteria = \tool_advancedreplace\file_search::get_criteria($record);
-        [$whereclause, $params] = \tool_advancedreplace\file_search::make_where_clause($criteria);
-        $totalfiles = $DB->count_records_select('files', $whereclause, $params);
+        // The sharding will be controlled by ranges of the id column of the mdl_files table.
+        // Here we implement a simple division. In future we could use WHERE clause to choose more even break points.
+
+        $maxid = $DB->get_field_sql('SELECT MAX(id) FROM {files}');
```

Comment:
This is something we agreed upon, but I'm still not convinced that splitting the shards like this is an improvement.

Would it be too inefficient to do a full count and have subsequent selects getting the max id with the where clause with the start id and a limit based on totalcount/numshards?

Reply:
Yes - I proposed doing all the counting in the master shard and giving the children id numbers to work with.  The split will be quite uneven as we have it.  The first 50 percent will be approximately empty. But this is actually not much of a disaster. We can just run it with a bigger number of shards. 

I don't think PR comments is the place to change the specs.

Reply:
I mainly brought it up because in my mind this is a pretty big regression when the number of processes running at a time are limited, especially if everything ends up being in one small section. 

But I agree it is as per the current spec and it's OK for our current purposes so I'm OK with merging.